department-of-veterans-affairs/caseflow
# Feature: Establishment Task during AMA Intake

Resolves #{[APPEALS-23288](https://jira.devops.va.gov/browse/APPEALS-23288)}

# Description
Please explain the changes you made here.

1.  Added `InitialTasksFactory#create_establishment_task` to the intake workflow
2. Created `EstablishmentTask` class
3.  Updated `TaskRows.jsx` to render the `EstablishmentTask` properly in the Case Timeline.

## Acceptance Criteria

## Testing Plan
AC 1:
1. Switch user to BVADWISE
2. Start an intake for form 10182
3. Enter the ID for a veteran with no prior decisions from VBMS (in my case 100000011)
5.  Fill out the form and click "Continue to next step"
6. Add 4 issues. One with MST, one with PACT, one with both designations, and one with neither designation.
7. Click "Establish appeal"
8. Now search cases for the veteran (100000011)
9. Select the case you just established from the docket.
10. Scroll down to the CaseTimeline and click "View Task Instructions"
11. You should see

- 1 EstablishmentTask completed
- Date completed on
- Assignee
- Completed by
- 4 added issues with the corresponding descriptions and special issues for each.

AC 2 & 3:
1. Switch user to BVADWISE
2. Start an intake for form 10182
3. Enter the ID for a veteran with a prior decision from VBMS (in my case 245620198)
5.  Fill out the form and click "Continue to next step" 
6. Add 3 issues.

- On the first, select "Higher Level Review was denied" and click next.
-  On the 2nd, select "Higher Level Review was denied", click the MST checkbox and click next
- On the 3nd, select "Higher Level Review was denied", click the PACT checkbox and click next
6. Click "Establish Appeal"
7. Search the case you just created.
8. Scroll down to the CaseTimeline and click "View Task Instructions"
9. You should see

- 1 EstablishmentTask completed
- Date completed on
- Assignee
- Completed by
- 3 added issues with the corresponding descriptions and special issues for each.
- On the 2nd and 3rd issue, you should see the original special issues from VBMS as well as the updated special issues added by the user in Intake.


# Frontend
## User Facing Changes
AC 1:
![Screenshot 2023-06-01 at 2 48 12 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/93014155/7c9768e9-2af9-432e-a8a8-c9dd2c7b5d97)
AC 2 & 3:
![Screenshot 2023-06-01 at 2 47 30 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/93014155/3cd7845c-f6e7-45b1-9fbd-a3f4e6440194)



## Storybook Story
*For Frontend (Presentationa) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
